src/utils/getTargetEngines.js

```diff
@@ -0,0 +1,89 @@
+const browserslist = require('browserslist');
+const semver = require('semver');
+
+/**
+ * Loads target node and browser versions from the following locations:
+ *   - package.json engines field
+ *   - package.json browserslist field
+ *   - browserslist or .browserslistrc files
+ *   - .babelrc or .babelrc.js files with babel-preset-env
+ */
+async function getTargetEngines(asset, isTarget, path) {
+  let targets = {};
+  let pkg = await asset.getConfig(['package.json'], {path});
+
+  // Only use engines in node_modules, not the target app
+  let engines = pkg && !isTarget ? pkg.engines : null;
+  let nodeVersion = engines && getMinSemver(engines.node);
+  if (typeof nodeVersion === 'string') {
+    targets.node = nodeVersion;
+  }
+
+  if (
+    engines &&
+    (typeof engines.browsers === 'string' || Array.isArray(engines.browsers))
+  ) {
+    targets.browsers = engines.browsers;
+  } else if (pkg && pkg.browserslist) {
+    targets.browsers = pkg.browserslist;
+  } else {
+    let browserslist = await loadBrowserslist(asset, path);
+    if (browserslist) {
+      targets.browsers = browserslist;
+    } else {
+      let babelTargets = await loadBabelrc(asset, path);
+      Object.assign(targets, babelTargets);
+    }
+  }
+
+  if (Object.keys(targets).length === 0) {
+    return null;
+  }
+
+  if (targets.browsers) {
+    targets.browsers = browserslist(targets.browsers).sort();
+  }
+
+  return targets;
+}
+
+function getMinSemver(version) {
```

Comment:
Related: https://github.com/npm/node-semver/pull/208. This is actually a pretty tricky thing to do.

Reply:
Yep. I think what I have here might be good enough for what we're using it for though, which is only parsing the `package.engines.node` field. It's unlikely to contain any super complex semver versions in there.